source/val/validation_state.cpp

```diff
@@ -1319,5 +1319,9 @@ bool ValidationState_t::IsValidStorageClass(
   return true;
 }
 
+std::string ValidationState_t::VkErrorID(std::string id) const {
```

Comment:
I think this should just take a uint32_t and construct the message here.

```
uint32_t ValidationState_t::VkErrorID(uint32_t id) const {
  if (!spvIsVulkanEnv(context_->target_env))
    return "";

  switch (id) {
    case 4425:
      return "VUID-WorkgroupSize-WorkgroupSize-04425";
    ...
    case 0: // an invalid id.
    default:
     return "";
}
```

Reply:
1. my worry is the switch statement becomes this awful 200 case long, the indirect lookup table
2. It is technically possible to have `VUID-x-0001` and `VUID-y-0001` if both are the same VU in the Vulkan spec using the "common validity" generation. Not an issue for the Built-in, but worry about when one case of that happens

Reply:
To 1, there are solutions. The speed is mostly irrelevant since we're only doing this if we're stopping because we found an error.

To 2, that sounds like a bad design, but I can't really change it. Mostly I would like the code to be briefer as these are rather long (and seemingly redundant) tags.

Reply:
1. agree

2. The reason is for things like all the `vkCmdDraw*` commands it was easier to write the VU once and was simpler for the VU generation logic to us the same number as the string was the unique part, not the number... but again, out of scope

Thoughts on

```
uint32_t ValidationState_t::VkErrorID(uint32_t id, const char* reference = nullptr) const;
```

This way if there is a overlap of the id number, there is something it if/else in the switch case

Reply:
The extra parameter seems reasonable. We can always revisit this later.